arithmetization/src/main/java/net/consensys/linea/zktracer/module/oob/OobChunk.java

```diff
@@ -0,0 +1,1294 @@
+import net.consensys.linea.zktracer.module.oob.parameters.ModexpCallDataSizeParameters;
+import net.consensys.linea.zktracer.module.oob.parameters.ModexpExtractParameters;
+import net.consensys.linea.zktracer.module.oob.parameters.ModexpLeadParameters;
+import net.consensys.linea.zktracer.module.oob.parameters.ModexpPricingParameters;
+import net.consensys.linea.zktracer.module.oob.parameters.ModexpXbsParameters;
+import net.consensys.linea.zktracer.module.oob.parameters.OobParameters;
+import net.consensys.linea.zktracer.module.oob.parameters.PrecompileCommonOobParameters;
+import net.consensys.linea.zktracer.module.oob.parameters.ReturnDataCopyOobParameters;
+import net.consensys.linea.zktracer.module.oob.parameters.SstoreOobParameters;
+import net.consensys.linea.zktracer.module.wcp.Wcp;
+import net.consensys.linea.zktracer.opcode.OpCode;
+import net.consensys.linea.zktracer.types.EWord;
+import net.consensys.linea.zktracer.types.UnsignedByte;
+import org.apache.tuweni.bytes.Bytes;
+import org.hyperledger.besu.datatypes.Address;
+import org.hyperledger.besu.evm.account.Account;
+import org.hyperledger.besu.evm.frame.MessageFrame;
+import org.hyperledger.besu.evm.internal.Words;
+
+@Getter
+public class OobChunk extends ModuleOperation {
+  private BigInteger oobInst;
+
+  private boolean isJump;
+  private boolean isJumpi;
+  private boolean isRdc;
+  private boolean isCdl;
+  private boolean isXCall;
+  private boolean isCall;
+  private boolean isCreate;
+  private boolean isSstore;
+  private boolean isDeployment;
+
+  private boolean isEcRecover;
+  private boolean isSha2;
+  private boolean isRipemd;
+  private boolean isIdentity;
+  private boolean isEcadd;
+  private boolean isEcmul;
+  private boolean isEcpairing;
+  private boolean isBlake2FCds;
+  private boolean isBlake2FParams;
+  private boolean isModexpCds;
+  private boolean isModexpXbs;
+  private boolean isModexpLead;
+  private boolean prcModexpPricing;
+  private boolean prcModexpExtract;
+
+  private boolean isModexpBbs;
+  private boolean isModexpEbs;
+  private boolean isModexpMbs;
+
+  private final boolean[] addFlag;
+  private final boolean[] modFlag;
+  private final boolean[] wcpFlag;
+
+  private final UnsignedByte[] outgoingInst;
+
+  private final BigInteger[] outgoingData1;
+  private final BigInteger[] outgoingData2;
+  private final BigInteger[] outgoingData3;
+  private final BigInteger[] outgoingData4;
+
+  private final BigInteger[] outgoingResLo;
+
+  private OobParameters oobParameters;
+
+  private BigInteger wghtSum;
+
+  private BigInteger precompileCost;
+
+  // Modules for lookups
+  private final Add add;
+  private final Mod mod;
+  private final Wcp wcp;
+
+  private final Hub hub;
+
+  private final int blake2FCallNumber;
+  private final int modexpCallNumber;
+
+  public OobChunk(
+      final MessageFrame frame,
+      final Add add,
+      final Mod mod,
+      final Wcp wcp,
+      final Hub hub,
```

Comment:
It seems we don't need to have the hub as a depence, neither oin iibChunk nor in OOB.java .. Better to remove it

Reply:
We use it here, but likely this part will die after the update. I will remove it later if needed.
```
 case CALL, CALLCODE, DELEGATECALL, STATICCALL:
        if (opCode == OpCode.CALL
            && !hub.pch().exceptions().stackUnderflow()
            && hub.pch().exceptions().any()) {
          isXCall = true;
          wghtSum = BigInteger.valueOf(0xCC);
        } else {
```